In [19]:
import pandas as pd 
import numpy as np 
import altair as alt
alt.data_transformers.enable("vegafusion")

DataTransformerRegistry.enable('vegafusion')

In [20]:
df = pd.read_csv('data/dataverse_files/Residential_1.csv')
df.head()

date  hour  energy_kWh
0  2012-06-01     1       1.011
1  2012-06-01     2       0.451
2  2012-06-01     3       0.505
3  2012-06-01     4       0.441
4  2012-06-01     5       0.468

In [21]:
df['date'] = pd.to_datetime(df['date'])
df['datetime'] = df['date'] + pd.to_timedelta(df['hour'] - 1, unit='h')

In [22]:
df_eda = alt.Chart(df).mark_line(color="#333").encode(
    x=alt.X("datetime:T", title="hour"),  
    y=alt.Y("energy_kWh:Q", title="Energy (kWh)")  
)

df_eda

alt.Chart(...)

In [ ]:
df06010212 = df[df['date'].isin(['2012-06-01', '2012-06-02'])]
# df06010212

/var/folders/tf/pzsl6f5j1wlcsbq4m08zvbtw0000gq/T/ipykernel_37638/1170177995.py:1: FutureWarning: The behavior of 'isin' with dtype=datetime64[ns] and castable values (e.g. strings) is deprecated. In a future version, these will not be considered matching by isin. Explicitly cast to the appropriate dtype before calling isin instead.
  df06010212 = df[df['date'].isin(['2012-06-01', '2012-06-02'])]


date  hour  energy_kWh            datetime
0  2012-06-01     1       1.011 2012-06-01 00:00:00
1  2012-06-01     2       0.451 2012-06-01 01:00:00
2  2012-06-01     3       0.505 2012-06-01 02:00:00
3  2012-06-01     4       0.441 2012-06-01 03:00:00
4  2012-06-01     5       0.468 2012-06-01 04:00:00
5  2012-06-01     6       0.451 2012-06-01 05:00:00
6  2012-06-01     7       0.464 2012-06-01 06:00:00
7  2012-06-01     8       0.897 2012-06-01 07:00:00
8  2012-06-01     9       1.066 2012-06-01 08:00:00
9  2012-06-01    10       0.946 2012-06-01 09:00:00
10 2012-06-01    11       0.465 2012-06-01 10:00:00
11 2012-06-01    12       0.792 2012-06-01 11:00:00
12 2012-06-01    13       0.468 2012-06-01 12:00:00
13 2012-06-01    14       0.674 2012-06-01 13:00:00
14 2012-06-01    15       0.503 2012-06-01 14:00:00
15 2012-06-01    16       0.556 2012-06-01 15:00:00
16 2012-06-01    17       0.925 2012-06-01 16:00:00
17 2012-06-01    18       0.713 2012-06-01 17:00:00
18 2012-06-01    19       0.597 2012-06-01 18:00:00
19 2012-06-01    20       0.559 2012-06-01 19:00:00
20 2012-06-01    21       0.752 2012-06-01 20:00:00
21 2012-06-01    22       1.584 2012-06-01 21:00:00
22 2012-06-01    23       4.188 2012-06-01 22:00:00
23 2012-06-02     0       0.613 2012-06-01 23:00:00
24 2012-06-02     1       0.442 2012-06-02 00:00:00
25 2012-06-02     2       0.496 2012-06-02 01:00:00
26 2012-06-02     3       0.484 2012-06-02 02:00:00
27 2012-06-02     4       0.447 2012-06-02 03:00:00
28 2012-06-02     5       0.460 2012-06-02 04:00:00
29 2012-06-02     6       0.500 2012-06-02 05:00:00
30 2012-06-02     7       0.696 2012-06-02 06:00:00
31 2012-06-02     8       0.627 2012-06-02 07:00:00
32 2012-06-02     9       0.961 2012-06-02 08:00:00
33 2012-06-02    10       0.982 2012-06-02 09:00:00
34 2012-06-02    11       0.840 2012-06-02 10:00:00
35 2012-06-02    12       1.159 2012-06-02 11:00:00
36 2012-06-02    13       2.213 2012-06-02 12:00:00
37 2012-06-02    14       0.457 2012-06-02 13:00:00
38 2012-06-02    15       0.479 2012-06-02 14:00:00
39 2012-06-02    16       0.462 2012-06-02 15:00:00
40 2012-06-02    17       0.473 2012-06-02 16:00:00
41 2012-06-02    18       2.104 2012-06-02 17:00:00
42 2012-06-02    19       1.536 2012-06-02 18:00:00
43 2012-06-02    20       1.152 2012-06-02 19:00:00
44 2012-06-02    21       0.935 2012-06-02 20:00:00
45 2012-06-02    22       0.871 2012-06-02 21:00:00
46 2012-06-02    23       0.942 2012-06-02 22:00:00

In [24]:
df06010212_eda = alt.Chart(df06010212).mark_line(color="#333").encode(
    x=alt.X("datetime:T", title="hour"),  
    y=alt.Y("energy_kWh:Q", title="Energy (kWh)")  
)
df06010212_eda

alt.Chart(...)

BC Hydro Time of Day Pricing (https://app.bchydro.com/accounts-billing/rates-energy-use/electricity-rates/residential-rates/time-of-day.html):
- 16:00 - 21:00 + 5 cent per kWh (peak)
- 21:00 - 23:00 + 0 cent per kWh 
- 23:00 - 07:00 - 5 cent per kWh (off-peak)
- 07:00 - 16:00 + 0 cent per kWh 

# Simple Simulation 1 
Assume:
- Home owner has a Tesla Model 3 Standard Range Plus of 50 kWh 
- The charging period over off peak is sufficient to supply all capacity needed 
- The energy output can match with energy demand 



In [27]:
saving_06010212 = df06010212[(df06010212['hour'] >= 16) & (df06010212['hour'] <= 20)]['energy_kWh'].sum()
saving_06010212

9.077

On 2012-06-01, and 2012-06-02, this household used 9.077 kWh of electricity in between 16:00 to 21:00. If the household is participated in BC Hydro's Time of Day Pricing and has a battery system, they can save up to 9.077 dollars. 

In [28]:
saving = df[(df['hour'] >= 16) & (df['hour'] <= 20)]['energy_kWh'].sum()
saving

9333.621

Same logic but extending the time frame to 2012-06-01 to 2015-10-03. This household can save up to 9333.621 dollars. 